buildconfig/CMake/BundleOSXConda.cmake

```diff
@@ -0,0 +1,45 @@
+macro(install_qt5_plugin_osx _qt_plugin_name _qt_plugins_var _prefix)
+  get_target_property(_qt_plugin_path "${_qt_plugin_name}" LOCATION)
+  if(EXISTS "${_qt_plugin_path}")
+    get_filename_component(_qt_plugin_file "${_qt_plugin_path}" NAME)
+    get_filename_component(_qt_plugin_type "${_qt_plugin_path}" PATH)
+    get_filename_component(_qt_plugin_type "${_qt_plugin_type}" NAME)
+    set(_qt_plugin_dest "${_prefix}/PlugIns/${_qt_plugin_type}")
+    install(
+      FILES "${_qt_plugin_path}"
+      DESTINATION "${_qt_plugin_dest}"
+      COMPONENT Runtime
+    )
+    set(${_qt_plugins_var}
+        "${${_qt_plugins_var}};\$ENV{DEST_DIR}\${CMAKE_INSTALL_PREFIX}/${_qt_plugin_dest}/${_qt_plugin_file}"
+    )
+  else()
+    message(FATAL_ERROR "QT plugin ${_qt_plugin_name} not found")
+  endif()
+endmacro()
+
+install(CODE "set(WORKBENCH_APP \"${WORKBENCH_APP}\")" COMPONENT Runtime)
+# Manually sort out the dependencies using cmake
```

Comment:
I want to get this PR through as it's great work on creating a package and I like we have everything in CMake rather than separate scripts.

Could I ask though that you add a few comments in a separate PR on the process here? For new comers I think the most confusing thing will be why the, at first glance random, list of targets have been selected. Thanks.